File: Escola/Cursos.Api/scripts/service/orgaoemissorService.js
///// <sumary>
///// By: Jeferson Braga do Carmo Create Date: 25/11/2015
///// Last Update: 11/13/2017 1:58:38 PM
///// Gerado automaticamente para alteracoes dos metodos comum alterar o template
///// </sumary>
"use strict";
app.factory("OrgaoemissorService", ["$http", "ngAuthSettings", function ($http, ngAuthSettings) {
        var service = {};
		var serviceBase = ngAuthSettings.apiServiceBaseUri;
        service.GetAll = function (callback) {
            $http.get(serviceBase + "Orgaoemissor/getall")
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.GetById = function (id, callback) {
            $http.get(serviceBase + "Orgaoemissor/getbyid/" + id)
               .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.Post = function (Orgaoemissor, callback) {
            $http.post(serviceBase + "Orgaoemissor/post", Orgaoemissor)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.Put = function (Orgaoemissor, callback) {
            $http.post(serviceBase + "Orgaoemissor/put", Orgaoemissor)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

		service.Delete = function (Orgaoemissor, callback) {
            $http.post(serviceBase + "Orgaoemissor/delete", Orgaoemissor)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        return service;
    }]);

File: Escola/Cursos.Repository/Services/OrgaoemissorService.cs
///// <sumary>
///// By: Jeferson Braga do Carmo Create Date: 25/11/2015
///// Last Update: 11/11/2017 9:23:24 PM

///// Para outros metodos implementar classe partial
///// </sumary>
using System;
using System.Linq;
using System.Collections.Generic;
using System.Threading.Tasks;
using Cursos.Repository.Services.Base;
using Cursos.Data;
using Cursos.Repository.Services.Interfaces;
using System.Linq.Expressions;
using Cursos.Repository.Work.Interface;
namespace  Cursos.Repository.Services
{
	  /// <summary>
      /// Public Service Class  OrgaoemissorService
      /// </summary>
      public partial class OrgaoemissorService : BaseServices, IOrgaoemissorService
      {

		public OrgaoemissorService(IUnitOfWork unitOfWork)
        {
			UnitOfWork = unitOfWork;
        }

		/// <summary>
        /// Add Range Entity AD_ORGAOEMISSOR
        /// </summary>
        /// <param name="entity"></param>
		public void Add(IEnumerable<AD_ORGAOEMISSOR> entity)
        {
            UnitOfWork.OrgaoemissorRepository.Add(entity);
        }

		/// <summary>
        /// Add Entity AD_ORGAOEMISSOR
        /// </summary>
        /// <param name="entity"></param>
		public void Add(AD_ORGAOEMISSOR entity)
        {	
            UnitOfWork.OrgaoemissorRepository.Add(entity);
        }
		/// <summary>
        /// Delete Entity AD_ORGAOEMISSOR
        /// </summary>
        /// <param name="entity"></param>
        public void Delete(AD_ORGAOEMISSOR entity)
        {
            UnitOfWork.OrgaoemissorRepository.Delete(entity);
        }
		 /// <summary>
        /// Get All Entity AD_ORGAOEMISSOR
        /// </summary>
        /// <returns>IEnumerable AD_ORGAOEMISSOR</returns>
        public IEnumerable<AD_ORGAOEMISSOR> GetAll()
        {
			return UnitOfWork.OrgaoemissorRepository.GetAll();
        }
	     
		/// <summary>
        /// Update Entity AD_ORGAOEMISSOR
        /// </summary>
        /// <param name="entity"></param>
        public void Update(AD_ORGAOEMISSOR entity)
        {
            UnitOfWork.OrgaoemissorRepository.Edit(entity);
        }

		/// <summary>
        /// Commit Transaction
        /// </summary>
        /// <returns></returns>
        public int Commit()
        {
            return UnitOfWork.Commit();
        }
		/// <summary>
        /// Commit Async Transaction
        /// </summary>
		public Task<int> CommitAsync()
        {
            return UnitOfWork.CommitAsync();
        }
				/// <summary>
        /// Not Implemented By AD_ORGAOEMISSOR Not Have Attribute Id in Entity
        /// </summary>
        /// <param name="id"></param>
        /// <returns>AD_ORGAOEMISSOR</returns>
		public AD_ORGAOEMISSOR GetById(int id)
        {
            throw new NotImplementedException();
        }
				
		public IEnumerable<AD_ORGAOEMISSOR> FindBy(Expression<Func<AD_ORGAOEMISSOR, bool>> predicate)
        {
            return UnitOfWork.OrgaoemissorRepository.FindBy(predicate);
        }
      }
}
